cmd/crc/cmd/start.go

```diff
@@ -156,7 +158,28 @@ func (s *startResult) prettyPrintTo(writer io.Writer) error {
 			"This cluster was built from OKD - The Community Distribution of Kubernetes that powers Red Hat OpenShift.",
 			"If you find an issue, please report it at https://github.com/openshift/okd"}, "\n"))
 	}
-	return err
+	if err != nil {
+		return err
+	}
+
+	if userType := config.Get(cmdConfig.AutoLoginAs).AsString(); len(userType) > 0 {
+		return s.runLoginCommandFor(userType)
```

Comment:
No error is returned/handled, and does not wait for the cluster to be available.

Reply:
I assumed the cluster is ready when this happens as it's at the same time the login info is printed.
Seems like this assumption is wrong :-/

Reply:
I am very concerned that this leads to unpreditability; the user gets no feedback about error or success in this case.

Reply:
https://github.com/code-ready/crc/blob/d8c8b8d5b7eda770a28ce66275d6e886b1b26550/pkg/crc/machine/start.go#L370

which might not be enough...

so this is followed by

https://github.com/code-ready/crc/blob/d8c8b8d5b7eda770a28ce66275d6e886b1b26550/pkg/crc/machine/start.go#L374-L377

doing:
https://github.com/code-ready/crc/blob/d8c8b8d5b7eda770a28ce66275d6e886b1b26550/pkg/crc/machine/kubeconfig.go#L33-L49

As you can see, this is mostly based on timeouts and retry. This is the technical debt I refer to.

Reply:
Thanks for explaining! That makes a lot of sense.
One option I could think about is to change the flag and not run login, but if the flag is set, set the context written to kubeconfig as current context at the same time?

Either way, how would you like to proceed with this? If you think having this option is not really viable feel free to close this, if you think it would be and we can get it in there good enough, I don't mind working to bring it up to standard.

Reply:
@praveenkumar WDYT?